File: 01-INTRODUCTIONS/08-ALGORITHMS/Examples.cs
using System;

namespace AlgorithmsExamples
{
    internal class Examples
    {
        static void Main(string[] args)
        {
            Console.WriteLine("===== ALGORITHM EXAMPLES =====\n");

            Example1_GreetUser();
            Example2_AddTwoNumbers();
            Example3_FindLargestNumber();
            Example4_EvenOrOdd();
            Example5_CalculateAverage();

            Console.WriteLine("\nAll examples completed.");
        }

        // Example 1
        static void Example1_GreetUser()
        {
            Console.WriteLine("EXAMPLE 1: Greeting Algorithm");

            string name = "Peyman";

            Console.WriteLine("Hello, " + name + "!");

            Console.WriteLine();
        }

        // Example 2
        static void Example2_AddTwoNumbers()
        {
            Console.WriteLine("EXAMPLE 2: Add Two Numbers");

            int number1 = 15;
            int number2 = 25;

            int sum = number1 + number2;

            Console.WriteLine($"{number1} + {number2} = {sum}");

            Console.WriteLine();
        }

        // Example 3
        static void Example3_FindLargestNumber()
        {
            Console.WriteLine("EXAMPLE 3: Find the Largest Number");

            int a = 18;
            int b = 42;

            if (a > b)
            {
                Console.WriteLine($"{a} is the largest number.");
            }
            else
            {
                Console.WriteLine($"{b} is the largest number.");
            }

            Console.WriteLine();
        }

        // Example 4
        static void Example4_EvenOrOdd()
        {
            Console.WriteLine("EXAMPLE 4: Even or Odd");

            int number = 17;

            if (number % 2 == 0)
            {
                Console.WriteLine($"{number} is even.");
            }
            else
            {
                Console.WriteLine($"{number} is odd.");
            }

            Console.WriteLine();
        }

        // Example 5
        static void Example5_CalculateAverage()
        {
            Console.WriteLine("EXAMPLE 5: Calculate Average");

            int grade1 = 90;
            int grade2 = 85;
            int grade3 = 95;

            double average = (grade1 + grade2 + grade3) / 3.0;

            Console.WriteLine($"Grades: {grade1}, {grade2}, {grade3}");
            Console.WriteLine($"Average: {average:F2}");

            Console.WriteLine();
        }
    }
}

/*
==========================================================
AUTHOR
==========================================================

Name: Peyman Miyandashti
University: Polytechnic University of Baja California
Program: Information Technology Engineering & Digital Innovation
Origin: IRAN (Mexico)
Year: 2026
*/

File: 01-INTRODUCTIONS/10-KOTLIN/01-INTRODUCTION-TO-KOTLIN/Examples.kt
/*
 * File: 10-KOTLIN/01-INTRODUCTION-TO-KOTLIN/Examples.kt
 * Type: Examples
 *
 * Author: Peyman Miyandashti
 * University: Polytechnic University of Baja California
 * Program: Information Technology Engineering & Digital Innovation
 * Origin: Iran (Mexico)
 * Year: 2026
 */

package introductiontokotlin

/*
=====================================================
EXAMPLE 1
The First Kotlin Program
=====================================================
*/

fun main() {
    println("Hello, World!")
}

/*
Output:
Hello, World!
*/


/*
=====================================================
EXAMPLE 2
Printing Multiple Lines
=====================================================
*/

fun multipleLinesExample() {
    println("Welcome to Kotlin!")
    println("I am learning Kotlin with IntelliJ IDEA.")
    println("Programming is fun.")
}

/*
Output:
Welcome to Kotlin!
I am learning Kotlin with IntelliJ IDEA.
Programming is fun.
*/


/*
=====================================================
EXAMPLE 3
Understanding fun
=====================================================
*/

fun sayHello() {
    println("Hello from a function!")
}

/*
Calling the function:

sayHello()

Output:
Hello from a function!
*/


/*
=====================================================
EXAMPLE 4
Understanding main()
=====================================================
*/

fun mainExample() {
    println("Programs start from main().")
}

/*
Explanation:

main() is the entry point of a Kotlin application.
The JVM starts executing the program here.
*/


/*
=====================================================
EXAMPLE 5
Using println()
=====================================================
*/

fun printlnExample() {
    println("This text appears on the console.")
}

/*
Output:
This text appears on the console.
*/


/*
=====================================================
EXAMPLE 6
Common Mistake: Misspelled println
=====================================================
*/

// Wrong
// prntln("Hello")

// Correct
fun correctedPrintln() {
    println("Hello")
}


/*
=====================================================
EXAMPLE 7
Common Mistake: Missing Parentheses
=====================================================
*/

// Wrong
// println "Hello"

// Correct
fun correctedParentheses() {
    println("Hello")
}


/*
=====================================================
EXAMPLE 8
Common Mistake: Extra Braces
=====================================================
*/

// Wrong
/*
fun brokenProgram() {
    println("Hello")
}}
*/

// Correct

fun fixedProgram() {
    println("Hello")
}


/*
=====================================================
EXAMPLE 9
Personalized Greeting
=====================================================
*/

fun greetingExample() {
    println("Hello Peyman!")
    println("Welcome to Kotlin 2026.")
}

/*
Output:
Hello Peyman!
Welcome to Kotlin 2026.
*/


/*
=====================================================
EXAMPLE 10
Motivational Example
=====================================================
*/

fun motivationExample() {
    println("Every expert was once a beginner.")
    println("Keep practicing Kotlin every day.")
}

/*
Output:
Every expert was once a beginner.
Keep practicing Kotlin every day.
*/


/*
END OF EXAMPLES
*/
